starlite/kwargs.py

```diff
@@ -50,6 +64,18 @@ def merge_parameter_sets(first: Set[ParameterDefinition], second: Set[ParameterD
     return result
 
 
+def is_array(model_field: ModelField):
+    return model_field.shape in [
+        SHAPE_LIST,
+        SHAPE_SET,
+        SHAPE_SEQUENCE,
+        SHAPE_TUPLE,
+        SHAPE_TUPLE_ELLIPSIS,
+        SHAPE_DEQUE,
+        SHAPE_FROZENSET,
+    ]
```

Comment:
I was going to suggest defining a set in the module scope to speed up the lookup times, but I felt like getting my nerd on:)

This profiles constructing a literal list inside a function and checking if the first element is in the list:
```
$ python -m timeit -s "def f(): 0 in [0,1,2,3,4,5,6]" "f()"
5000000 loops, best of 5: 60.3 nsec per loop
```
This profiles constructing the list as a global and checking for the first element (slower!):
```
$ python -m timeit -s "L = list(range(7))" -s "def f(): 0 in L" "f()"
5000000 loops, best of 5: 69.1 nsec per loop
```
So the construction of the list literal must get optimized away and its more expensive to look up a value from the global scope than the function local scope.

Where the list falls over is for elements other than the first:
```
$ python -m timeit -s "def f(): 6 in [0,1,2,3,4,5,6]" "f()"
2000000 loops, best of 5: 110 nsec per loop
```

I suppose the right thing to do is to use a set literal inside the function as it will have faster lookup times for all but the first element of the list:
```
$ python -m timeit -s "def f(): 6 in {0,1,2,3,4,5,6}" "f()"
5000000 loops, best of 5: 65 nsec per loop
```

... but these are very small numbers indeed.

Reply:
A set in the global scope sounds best tbh.